scripts/convert_bundle_v2.py

```diff
@@ -0,0 +1,149 @@
+    return {
+        repo: {
+            part: os.path.join(parent_dir, f'{uuid}.{part}{ext}.json')
+            for part in ('manifest', 'metadata')
+        }
+        for repo, ext in (('dss', ''), ('tdr', '.tdr'))
+    }
+
+
+def find_concrete_type(bundle: Bundle, entity_id: EntityID):
+    entity_key = one(k for k, v in bundle.metadata_files.items()
+                     if k != 'links.json' and v['provenance']['document_id'] == entity_id)
+    return bundle.concrete_entity_type(entity_key)
+
+
+def convert_version(version):
+    return datetime.strptime(
+        version,
+        dss.version_format
+    ).strftime(
+        tdr.Plugin.timestamp_format
+    )
+
+
+def convert_bundle(bundle: Bundle, source_id: str) -> None:
+    links_json = bundle.metadata_files['links.json']
+    links_json['schema_type'] = 'links'  # DCP/1 uses 'link_bundle'
+    for link in links_json['links']:
+        process_id = link.pop('process')
+        link['process_id'] = process_id
+        link['process_type'] = find_concrete_type(bundle, process_id)
+        link['link_type'] = 'process_link'  # No supplementary files in DCP/1 bundles
+        for component in ('input', 'output'):  # Protocols already in desired format
+            del link[f'{component}_type']  # Replace abstract type with concrete types
+            component_list = link[f'{component}s']
+            component_list[:] = [
+                {
+                    f'{component}_id': component_id,
+                    f'{component}_type': find_concrete_type(bundle, component_id)
+                }
+                for component_id in component_list
+            ]
+
+    for entry in bundle.manifest:
+        entry['version'] = convert_version(entry['version'])
+        if entry['indexed']:
+            entity_json = bundle.metadata_files[entry['name']]
+            # Size of the entity JSON in TDR, not the size of pretty-printed
+            # output file.
+            entry['size'] = len(json.dumps(entity_json).encode('UTF-8'))
+            # Only include mandatory checksums
+            del entry['sha1']
+            del entry['s3_etag']
+            entry['crc32c'] = ''
+            entry['sha256'] = ''
+        else:
+            # Generate random DRS ID
+            entry['drs_path'] = f'v1_{source_id}_{uuid.uuid4()}'
+
+        if entry['name'] == 'links.json':
+            # links.json has no FQID of its own in TDR since its FQID is used
+            # for the entire bundle.
+            entry['uuid'] = bundle.uuid
+            assert entry['version'] == bundle.version
+
+
+def main(argv):
+    """
+    Load a canned bundle from DCP/1 and write *.manifest.tdr and *.metadata.tdr
+    files showing the desired output for DCP/2.
+    """
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--bundle-uuid', '-b',
+                        help='The UUID of the existing DCP/1 canned bundle.')
+    parser.add_argument('--source-uuid', '-s',
```

Comment:
```suggestion
    parser.add_argument('--source-uuid', '-s',
                        default='cafebabe-feed-4bad-dead-beaf8badf00d',
```

Reply:
Personally I'd prefer that this also be required, we'll see what Hannes thinks.